ops/testing.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Infrastructure to build unit tests for charms using the ops library."""
+
+from ._private.harness import (
+    ActionFailed,
+    ActionOutput,
+    AppUnitOrName,
+    CharmBase,
+    CharmMeta,
+    CharmType,
+    Container,
+    ExecArgs,
+    ExecHandler,
+    ExecProcess,
+    ExecResult,
+    Harness,
+    ReadableBuffer,
+    RelationNotFoundError,
+    RelationRole,
+    YAMLStringOrFile,
+    charm,
+    framework,
+    model,
+    pebble,
+    storage,
+)
+
+# The Harness testing framework.
+_ = ActionFailed
+_ = ActionOutput
+_ = AppUnitOrName
+_ = CharmType
+_ = ExecArgs
+_ = ExecHandler
+_ = ExecResult
+_ = Harness
+_ = ReadableBuffer
+_ = YAMLStringOrFile
+
+# Names exposed for backwards compatibility
+_ = CharmBase
```

Comment:
I guess they don't hurt too much, but it is a pity we need these here. Have you tested using super-tox without them? If they're used, which ones?

Reply:
I haven't, partly because there is definitely breakage on ~12 charms because they're using `_TestingModelBackend`, so I would need to work out if it's additional breakage or only that.

I still feel like the right-thing-to-do-(tm) is to expose these, just like we did with `main` and `model`, even if we don't know of any specific cases where it would break things, because the API was that these names were available, even if we really hope or expect that no-one was using that.

If the team (or team leader :smile:) strongly prefers that we don't have these, then I can spend the time to figure out if there will be known breakage.

Reply:
Yep, that's fair enough -- let's leave it as is.